main_js.go

```diff
@@ -898,382 +921,388 @@ func (pe *pcdeditor) runImpl(ctx context.Context) error {
+					switch e.Code {
+					case "KeyW":
+						pe.vi.Move(0.05, 0, 0)
+					case "KeyA":
+						pe.vi.Move(0, 0.05, 0)
+					case "KeyS":
+						pe.vi.Move(-0.05, 0, 0)
+					case "KeyD":
+						pe.vi.Move(0, -0.05, 0)
+					case "KeyQ":
+						pe.vi.Move(0, 0, 0.02)
+					case "KeyE":
+						pe.vi.Move(0, 0, -0.02)
+					}
+				case "BracketRight", "Backslash":
+					switch e.Code {
+					case "BracketRight":
+						pe.vi.IncreaseFOV()
+					case "Backslash":
+						pe.vi.DecreaseFOV()
+					}
+				case "F10":
+					pe.cmd.Crop()
+				case "F11":
+					pe.vi.SnapYaw()
+				case "F12":
+					pe.vi.SnapPitch()
+				case "KeyP":
+					vib3D = !vib3D
 				}
-			case "KeyU":
-				pe.cmd.Undo()
-			case "KeyF":
-				pe.cmd.AddSurface(defaultResolution)
-			case "KeyV", "KeyH":
-				switch e.Code {
-				case "KeyV":
-					pe.cmd.SnapVertical()
-				case "KeyH":
-					pe.cmd.SnapHorizontal()
-				}
-			case "ArrowUp", "ArrowDown", "ArrowLeft", "ArrowRight", "PageUp", "PageDown":
-				var dx, dy, dz float32
-				switch e.Code {
-				case "ArrowUp":
-					dy = 0.05
-				case "ArrowDown":
-					dy = -0.05
-				case "ArrowLeft":
-					dx = -0.05
-				case "ArrowRight":
-					dx = 0.05
-				case "PageUp":
-					dz = 0.05
-				case "PageDown":
-					dz = -0.05
-				}
-				s, c := math.Sincos(pe.vi.yaw)
-				pe.cmd.TransformCursors(mat.Translate(
-					float32(c)*dx-float32(s)*dy,
-					float32(s)*dx+float32(c)*dy,
-					dz,
-				))
-			case "Home", "End":
-				var dyaw float32
-				switch e.Code {
-				case "Home":
-					dyaw = 0.005
-				case "End":
-					dyaw = -0.005
-				}
-				center := pe.cmd.RectCenterPos()
-				pe.cmd.TransformCursors(
-					mat.Translate(center[0], center[1], center[2]).
-						Mul(mat.Rotate(0, 0, 1, dyaw)).
-						Mul(mat.Translate(-center[0], -center[1], -center[2])),
-				)
-			case "KeyW", "KeyA", "KeyS", "KeyD", "KeyQ", "KeyE":
-				switch e.Code {
-				case "KeyW":
-					pe.vi.Move(0.05, 0, 0)
-				case "KeyA":
-					pe.vi.Move(0, 0.05, 0)
-				case "KeyS":
-					pe.vi.Move(-0.05, 0, 0)
-				case "KeyD":
-					pe.vi.Move(0, -0.05, 0)
-				case "KeyQ":
-					pe.vi.Move(0, 0, 0.02)
-				case "KeyE":
-					pe.vi.Move(0, 0, -0.02)
-				}
-			case "BracketRight", "Backslash":
-				switch e.Code {
-				case "BracketRight":
-					pe.vi.IncreaseFOV()
-				case "Backslash":
-					pe.vi.DecreaseFOV()
-				}
-			case "F10":
-				pe.cmd.Crop()
-			case "F11":
-				pe.vi.SnapYaw()
-			case "F12":
-				pe.vi.SnapPitch()
-			case "KeyP":
-				vib3D = !vib3D
-			}
-		case <-pe.chContextLost:
-			return errContextLostEvent
-		case <-tick.C:
-			if vib3D {
-				if vib3DX < 0 {
-					vib3DX = vib3DXAmp
+			case <-pe.chContextLost:
+				return errContextLostEvent
+			case <-tick.C:
+				if vib3D {
+					if vib3DX < 0 {
+						vib3DX = vib3DXAmp
+					} else {
+						vib3DX = -vib3DXAmp
+					}
 				} else {
-					vib3DX = -vib3DXAmp
+					vib3DX = 0
+					continue L_INPUT // process input again without rendering
```

Comment:
The diff is a bit hard to read here but I assume the only real change is `continue L_INPUT`.

Reply:
You can hide whitespace from diff
![image](https://github.com/user-attachments/assets/2e89e14c-9b16-42ba-844b-ceb52b3b00ce)